packages/@romejs/core/master/commands/ci.ts

```diff
@@ -5,10 +5,31 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import {commandCategories, createMasterCommand} from '../../commands';
+import {DiagnosticsPrinter} from '@romejs/cli-diagnostics';
 import {MasterRequest} from '@romejs/core';
-import test from './test';
+import {commandCategories, createMasterCommand} from '../../commands';
 import lint from './lint';
+import test from './test';
+
+async function runChildCommand(
+  req: MasterRequest,
+  fn: () => Promise<void>,
+): Promise<void> {
+  try {
+    await fn();
+    throw new Error('Expected a printer');
+  } catch (err) {
+    if (err instanceof DiagnosticsPrinter) {
+      if (err.hasDiagnostics()) {
+        throw err;
+      } else {
+        req.master.handleRequestError(req, err);
```

Comment:
This is kind of dirty. Is this the right approach?

Reply:
Maybe the following would be cleaner:
```suggestion
        err.print();
```

Reply:
That would actually be a noop when there's no diagnostics. There's manual calls to `printer.footer()` which is what prints the "No errors found" message, is that what you meant?

Reply:
Yeah, I called `handleRequestError` to keep the `bridge.alive` test and the `footer` call. The logic could be c/p. Feel free to update the branch to change this.